massgov/openmass
# DP-17535 Remove ability to create document on IEF form in List item document paragraph

**Description:**
The "List item document" paragraph which appears on the Curated List node has an entity reference field that references document media entities. This field uses the inline-entity-form widget to reference these entities. I was able to uncheck "create new media entities" on the widget settings on this page `/admin/structure/paragraphs_type/list_item_document/form-display` but [this patch](https://www.drupal.org/project/inline_entity_form/issues/2913571) was needed to allow disabling of editing of referenced entities.

As a note this work was done on the "List item document" paragraph and not specifically on the Curated List node. So this change will take effect everywhere this paragraph is used which includes other paragraphs and node types. If we think this is not acceptable or may cause a problem lets close this PR and think of another way to handle this (maybe with CSS or something?). 

Patch used: https://www.drupal.org/project/inline_entity_form/issues/2913571

**Jira:**
https://jira.mass.gov/browse/DP-17535

**To Test:**

1. - [ ] Edit a curated list node
2. - [ ] Add or edit a "List - Manual" paragraph
3. - [ ] Verify you can add/remove the document media entities under List Item Documents 
4. - [ ] Verify you can NOT create/edit the document media entities under List Item Documents 
5. - [ ] Verify the forms save correctly
6. - [ ] Verify the documents display as expected on the page





**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)
